### PR TITLE
moved labels to the same device as logits for LILT model

### PR DESCRIPTION
**Pull request by sushmanthreddy**
# What does this PR do?

As suggested in the [#22561](https://github.com/huggingface/transformers/issues/22561)  moved labels to the same device as logits for the lilt model.

@sgugger pls review and merge it in the main branch.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks a lot!
